apps/src/lib/node/ledger/shims/abcipp_shim.rs

```diff
@@ -181,31 +213,101 @@ impl AbcippShim {
     }
 }
 
+/// Indicates how [`AbciService`] should
+/// check whether or not it needs to take
+/// action.
+#[derive(Debug)]
+enum CheckAction {
+    /// No check necessary.
+    NoAction,
+    /// Check a given block height.
+    Check(i64),
+    /// The action been taken.
+    AlreadySuspended,
+}
+
 #[derive(Debug)]
 pub struct AbciService {
+    /// A channel for forwarding requests to the shell
     shell_send: std::sync::mpsc::Sender<(
         Req,
         tokio::sync::oneshot::Sender<Result<Resp, BoxError>>,
     )>,
+    /// Indicates if the consensus connection is suspended.
+    suspended: bool,
+    /// This resolves the non-completing futures returned to tower-abci
+    /// during suspension.
+    pub shutdown: broadcast::Sender<()>,
```

Comment:
I think this can now be made private, right?
```suggestion
    shutdown: broadcast::Sender<()>,
```